train.py

```diff
@@ -400,6 +400,11 @@ def loss_fn(pred, labels):
             optimizers.step()
             lr_schedulers.step()
 
+            if job_config.experimental.offload_activations:
+                # NOTE: We need `gc.collect` to ensure that CPU memory is freed
```

Comment:
This needs some more investigation. Maybe there is some ref cycle that I am not aware of.